clients/js/src/interfaces.ts

```diff
@@ -77,3 +77,8 @@ export async function undefer<T>(obj: Deferrable<T>): Promise<T> {
     await Promise.all(Object.entries(obj).map(async ([k, v]) => [k, await v])),
   );
 }
+
+export type UpstreamProvider =
+  | EIP1193Provider
```

Comment:
Think this needs EIP1193 Signer as well.

Reply:
The EIP-1193 interface is just the interface used to communicate with the Web3 wallet, make JSON-RPC requests, event notification etc. - so all operations are accessed through it, rather than having separate Provider/Signer interfaces like with Ethers.

See: https://eips.ethereum.org/EIPS/eip-1193

Reply:
What confused me was `Ethers5Signer` listed as one of the options of `UpstreamProvider`.

Reply:
The Ethers5Signer / Ethers5Provider types were extracted from the ethers package so as to not require importing both ethers v5 and ethers v6. For Ethers it needs to interact with them an entirely different and more complex way than with EIP-1193 compatible providers.